### PR TITLE
Issue/15

### LINKED ISSUE #15: Implement multiple configurations - RabbitMQClientConfig
Hello, first of all thank you very much for all the hard work you're making with this Rabbitmq adapter for quarkus, I was reading the code and I want to know if is possible to add support to have multiple sources for the **RabbitMQClientConfig** class. Rigth now the RabbitMQClient has the following bean creation.

```
RabbitMQClientImpl(RabbitMQClientConfig config, TlsConfig tlsConfig, ManagedExecutor managedExecutor) {
        this.config = config;
        this.tlsConfig = tlsConfig;
        this.managedExecutor = managedExecutor;
        this.connections = new HashMap();
    }
```

I want to know if is possible to implement something like this 

```
RabbitMQClientImpl(Map<String, RabbitMQClientConfig> config, TlsConfig tlsConfig, ManagedExecutor managedExecutor) {
        this.config = config;
        this.tlsConfig = tlsConfig;
        this.managedExecutor = managedExecutor;
        this.connections = new HashMap();
    }
```

Using this approach several instance of RabbitMQClientConfig can be created using this method:

```
public RabbitMQClientConfig getProperty(String name){
    String key = "quarkus.rabbitmqclient";
    if(Objects.nonNull(name) && !name.isEmpty() && !name.isBlank()){
      key += "." + name;
    }
    return ConfigProvider.getConfig().getValue(key, RabbitMQClientConfig.class);
}
```

instead of using this:

```
@ConfigRoot(name = "rabbitmqclient", phase = ConfigPhase.RUN_TIME)
public class RabbitMQClientConfig { ... }
```

and when the method `RabbitMQClient#connect(String name)` is called then the `RabbitMQClientConfig` used could be pointing to the quarkus.rabbitmqclient.config-name property.

If you see as something that can be implemented please, I wanna read comments and your plan so you can guide me to implement something to cover this use case.

### PR DESCRIPTION
**Pull request by bpasson**
Fixes #15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,6 +87,30 @@ public class MessageService {
...
+The extension supports having multiple RabbitMQ clients. You can add named RabbitMQ clients as follows.
+
+```properties
+quarkus.rabbitmqclient.<name>.virtual-host=/
+quarkus.rabbitmqclient.<name>.username=<username>
+quarkus.rabbitmqclient.<name>.password=<password>
+quarkus.rabbitmqclient.<name>.hostname=localhost
+quarkus.rabbitmqclient.<name>.port=5672
+```
+All configuration options that are available on de default non named RabbitMQ client are available. Injecting a named RabbitMQ client, e.g. foo, can be achieved as follows.
```
Comment (loicmathieu) [inline README.md:100]:
```suggestion
All configuration options that are available on the default non named RabbitMQ client are available. Injecting a named RabbitMQ client, e.g. foo, can be achieved as follows.
```

**Review by loicmathieu — APPROVED** [review]
I spotted a small typo and add some comments but generaly it's OK for me.

File: runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClients.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public RabbitMQClient getRabbitMQClient(String name) {
+        if (name == null) {
+            if (defaultClient == null) {
+                defaultClient = new RabbitMQClientImpl(rabbitMQClientsConfig.defaultClient, tlsConfig, managedExecutor);
+            }
+            return defaultClient;
+        } else {
+            return namedClients.computeIfAbsent(name,
+                    n -> new RabbitMQClientImpl(rabbitMQClientsConfig.namedClients.get(n), tlsConfig, managedExecutor));
+        }
```
[the remark this answers is not in the document]
Comment (bpasson) [reply runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClients.java:56]: There is no risk of duplicate, but as RabbitMQClients is basically a bean you can inject it in code and then call the `getRabbitMQClient(name)` method. This way I can make sure if someone goes that road the existing client is returned. I could handle that with a collection, but then I would have to loop through a collection and add a name property to the RabbitMQClient to find the proper client.

File: runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClients.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    public void destroy() {
+        // check for null, the producer method might never have been called.
+        if (defaultClient != null) {
+            defaultClient.disconnect();
+        }
+        namedClients.forEach((k, v) -> {
+            v.disconnect();
+        });
+    }
```
[the remark this answers is not in the document]
Comment (bpasson) [reply runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClients.java:75]: True, though we have the collection anyways as the RabbitMQClients bean can also be used to retrieve clients. Not sure why anyone would ever want to do that, but it is possible.

**Pull request merged**